<?xml version="1.0"?>
<launch>
    <group ns="drive_to_object">

        <node name="drive_to_object" pkg="drive_to_object" type="drive_to_object_node" output="screen"/>
        <rosparam command="load" file="$(find drive_to_object)/config/drive_to_object.yaml"/>
        
        <node name="x_position_pid" pkg="pid" type="controller" >
            <param name="node_name" value="x_position_pid" />
            <param name="Kp" value="1.0" />
            <param name="Ki" value="0.0" />
            <param name="Kd" value="0.0" />
            <param name="upper_limit" value="4.5" />
            <param name="lower_limit" value="-4.5" />
            <param name="windup_limit" value="10" />
            <param name="max_loop_frequency" value="250.0" />
            <param name="min_loop_frequency" value="250.0" />
            <param name="topic_from_controller" value="x_position_pid/x_command" />
            <param name="setpoint_topic" value="x_position_pid/x_cmd_pub" />
            <param name="topic_from_plant" value="x_position_pid/x_state_pub" />
            <remap from="pid_debug" to="x_position_pid/pid_debug" />
            <remap from="pid_enable" to="x_position_pid/pid_enable" />
        </node>
        
        <node name="y_position_pid" pkg="pid" type="controller" >
            <param name="node_name" value="y_position_pid" />
            <param name="Kp" value="1.0" />
            <param name="Ki" value="0.0" />
            <param name="Kd" value="0.0" />
            <param name="upper_limit" value="4.5" />
            <param name="lower_limit" value="-4.5" />
            <param name="windup_limit" value="10" />
            <param name="max_loop_frequency" value="250.0" />
            <param name="min_loop_frequency" value="250.0" />
            <param name="topic_from_controller" value="y_position_pid/y_command" />
            <param name="setpoint_topic" value="y_position_pid/y_cmd_pub" />
            <param name="topic_from_plant" value="y_position_pid/y_state_pub" />
            <remap from="pid_debug" to="y_position_pid/pid_debug" />
            <remap from="pid_enable" to="y_position_pid/pid_enable" />
        </node>


    </group>
	<node pkg="tf2_ros" type="static_transform_publisher" name="baselink_to_intake" args="0.3 0 0 0 0 0 1 base_link intake  " />
	<node pkg="tf2_ros" type="static_transform_publisher" name="baselink_to_frontbumper" args="-0.64 0 0 0 0 1 0 base_link shooter_bumper  " />


</launch>
